server/gorm/client.go

```diff
@@ -301,3 +307,32 @@ func (c *Client) Run(ctx context.Context, q storage.Query) storage.Iterator {
 		return nil
 	}
 }
+
+func (c *Client) GetRecentSpecRevisions(ctx context.Context, offset int32, projectID, apiID, versionID string) storage.Iterator {
+	mylock()
+	defer myunlock()
+
+	op := c.db.Select("specs.*").
+		Table("specs").
+		Joins("JOIN (?) AS grp ON specs.project_id = grp.project_id AND specs.api_id = grp.api_id AND specs.version_id = grp.version_id AND specs.spec_id = grp.spec_id AND specs.revision_create_time = grp.recent_create_time",
+			c.db.Select("project_id, api_id, version_id, spec_id, MAX(revision_create_time) AS recent_create_time").
+				Table("specs").
+				Group("project_id, api_id, version_id, spec_id")).
+		Order("key").
+		Offset(int(offset)).
+		Limit(1000)
```

Comment:
Updated to 100,000 as a temporary fix until we can upgrade our iterators. I think the ideal solution is to have our iterators read only the requested number of resources (or a slightly larger amount), then automatically query for another page if the user of the iterator keeps requesting elements (as we might see while filtering).

Reading entire tables into memory isn't scalable and the work of running multiple queries seems like it would be much better abstracted away as part of the iterator.